Cleaner_42.sh

```diff
@@ -44,49 +44,88 @@ echo -e "\033[33m\n -- Available Storage Before Cleaning : || $Storage || --\033
-/bin/rm -rf "$HOME"/.zcompdump* &>/dev/null
-/bin/rm -rf "$HOME"/.cocoapods.42_cache_bak* &>/dev/null
-
-#Trash
-/bin/rm -rf "$HOME"/.Trash/* &>/dev/null
-
-#General Caches files
-#giving access rights on Homebrew caches, so the script can delete them
-/bin/chmod -R 777 "$HOME"/Library/Caches/Homebrew &>/dev/null
-/bin/rm -rf "$HOME"/Library/Caches/* &>/dev/null
-/bin/rm -rf "$HOME"/Library/Application\ Support/Caches/* &>/dev/null
-
-#Slack, VSCode, Discord and Chrome Caches
-/bin/rm -rf "$HOME"/Library/Application\ Support/Slack/Service\ Worker/CacheStorage/* &>/dev/null
-/bin/rm -rf "$HOME"/Library/Application\ Support/Slack/Cache/* &>/dev/null
-/bin/rm -rf "$HOME"/Library/Application\ Support/discord/Cache/* &>/dev/null
-/bin/rm -rf "$HOME"/Library/Application\ Support/discord/Code\ Cache/js* &>/dev/null
-/bin/rm -rf "$HOME"/Library/Application\ Support/discord/Crashpad/completed/*  &>/dev/null
-/bin/rm -rf "$HOME"/Library/Application\ Support/Code/Cache/* &>/dev/null
-/bin/rm -rf "$HOME"/Library/Application\ Support/Code/CachedData/* &>/dev/null
-/bin/rm -rf "$HOME"/Library/Application\ Support/Code/Crashpad/completed/* &>/dev/null
-/bin/rm -rf "$HOME"/Library/Application\ Support/Code/User/workspaceStorage/* &>/dev/null
-/bin/rm -rf "$HOME"/Library/Application\ Support/Google/Chrome/Profile\ [0-9]/Service\ Worker/CacheStorage/* &>/dev/null
-/bin/rm -rf "$HOME"/Library/Application\ Support/Google/Chrome/Default/Service\ Worker/CacheStorage/* &>/dev/null
-/bin/rm -rf "$HOME"/Library/Application\ Support/Google/Chrome/Profile\ [0-9]/Application\ Cache/* &>/dev/null
-/bin/rm -rf "$HOME"/Library/Application\ Support/Google/Chrome/Default/Application\ Cache/* &>/dev/null
-/bin/rm -rf "$HOME"/Library/Application\ Support/Google/Chrome/Crashpad/completed/* &>/dev/null
-
-#.DS_Store files
-find "$HOME"/Desktop -name .DS_Store -depth -exec /bin/rm {} \; &>/dev/null
-
-#tmp downloaded files with browsers
-/bin/rm -rf "$HOME"/Library/Application\ Support/Chromium/Default/File\ System &>/dev/null
-/bin/rm -rf "$HOME"/Library/Application\ Support/Chromium/Profile\ [0-9]/File\ System &>/dev/null
-/bin/rm -rf "$HOME"/Library/Application\ Support/Google/Chrome/Default/File\ System &>/dev/null
-/bin/rm -rf "$HOME"/Library/Application\ Support/Google/Chrome/Profile\ [0-9]/File\ System &>/dev/null
-
-#things related to pool (piscine)
-/bin/rm -rf "$HOME"/Desktop/Piscine\ Rules\ *.mp4
-/bin/rm -rf "$HOME"/Desktop/PLAY_ME.webloc
+
+should_log=0
+if [[ "$1" == "-p" || "$1" == "--print" ]]; then
+	should_log=1
+fi
+
+function clean_glob {
+	# don't do anything if argument count is zero (unmatched glob).
+	if [ -z "$1" ]; then
+		return 0
+	fi
+
+	if [ $should_log -eq 1 ]; then
+		for arg in "$@"; do
+			du -sh "$arg" 2>/dev/null
+		done
+	fi
+
+	/bin/rm -rf "$@" &>/dev/null
+
+	return 0
+}
+
+function clean {
+	# to avoid printing empty lines
+	# or unnecessarily calling /bin/rm
+	# we resolve unmatched globs as empty strings.
+	shopt -s nullglob
```

Comment:
does shopt work properly on macOS?

Reply:
i tested it in one of the clusters iMacs and it works as expected.